### DISCUSSION TITLE
Setting message_id

### DISCUSSION THREAD

**Discussion by kolawolebalogun** [python-smpplib/python-smpplib#153] [Q&A] [answered]
Please is it possible to set The unique message identifier, that will later be returned in the DLR for reference purpose.

Let me explain my current process below:

**_Our approach is we send a message to the smsc gateway we get an id from smsc, we save this id and when DLR comes we match the id from the dlr to the id we previously save (because we push million of messages for a short period this approach isn't optimal for us, the back and forth to the database).

what I will like to acheive is if there is a way we can add a parameter to the submit message to smsc that and retrieve this id when the dlr comes back._**

**Comment by code-of-kpp** [comment, marked as answer]
Please take a look at strategies concept developed at #6
